java/src/main/java/sh/platform/languages/SampleCodeType.java

```diff
@@ -0,0 +1,70 @@
+package sh.platform.languages;
+
+import com.google.gson.Gson;
+import sh.platform.languages.sample.MongoDBSample;
+import sh.platform.languages.sample.MySQLSample;
+import sh.platform.languages.sample.PostgreSQLSample;
+import sh.platform.languages.sample.RedisSample;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public enum SampleCodeType {
+
+    MONGODB(new MongoDBSample(), "MongoDB"),
+    MYSQL(new MySQLSample(), "MySQL"),
+    POSTGRESQL(new PostgreSQLSample(), "PostgreSQL"),
+    REDIS(new RedisSample(), "Redis");
```

Comment:
This may just be a Java thing, but this block confuses me.  Why is SampleCodeType an enum (which I assume is being used as a typedef), and why the extra layer of indirection here with alternate values?  This adds more work when adding additional examples.  Why not just make those classes subclasses of an abstract base, reference them directly by name with a `new` in the router switch, and be done with it?

Reply:
That is exactly what the JVM does to us.
There is a nice book, effective Java that talks about the power of enum in Java:

https://www.amazon.com.br/Effective-Java-3rd-Joshua-Bloch/dp/0134685997?tag=goog0ef-20&smid=A1ZZFT5FULY4LN&ascsubtag=go_1157433115_58530734048_257324212232_pla-433354950479_c_